airflow/models.py

```diff
@@ -341,18 +341,35 @@ def kill_zombies(self, session=None):
         from airflow.jobs import LocalTaskJob as LJ
         self.log.info("Finding 'running' jobs without a recent heartbeat")
         TI = TaskInstance
+        LDT = LastDeployedTime
         secs = configuration.getint('scheduler', 'scheduler_zombie_task_threshold')
         limit_dttm = datetime.utcnow() - timedelta(seconds=secs)
         self.log.info("Failing jobs without heartbeat after %s", limit_dttm)
+        for deploy_orphan in session.query(TI).join(
+            LJ, TI.job_id == LJ.id
+        ).join(
+            LDT, TI.job_id != LDT.last_deployed
+        ).filter(
+            TI.state == State.RUNNING
+        ).filter(
+            LJ.latest_heartbeat < LDT.last_deployed
+        ):
+            deploy_orphan.refresh_from_db()
+            if deploy_orphan and deploy_orphan.dag_id in self.dags:
+                dag = self.dags[deploy_orphan.dag_id]
+                task = dag.get_task(deploy_orphan.task_id)
+                deploy_orphan.task = task
+                deploy_orphan.handle_deploy_orphan()
```

Comment:
This is how orphans deploy orphans are detected - this is done right before we check for zombies, so unlimited downtime between deploys is possible without the adoption mechanism creating a problem.  There is no need for the "real" scheduler to clear these tasks out of the in-memory executor maps because deploy orphans are definitionally not in those maps.